internal/resource/nvml-device.go

```diff
@@ -99,3 +100,35 @@ func (d nvmlDevice) GetPCIClass() (uint32, error) {
 	}
 	return nvDevice.Class, nil
 }
+
+func (d nvmlDevice) IsImexCapable() (bool, error) {
+	return true, nil
+}
+
+func (d nvmlDevice) GetClusterUUID() (string, error) {
```

Comment:
Question: Are `GetClusterUUID()` and `GetCliqueID()` ever used separately, or do they represent an ID pair? Is a more generic function that returns both more relevant here -- especially since we would remove the need to call `GetGpuFabricInfo()` twice?

Reply:
we could return the entire `GpuFabricInfo` and simply process the needed values at the GFD level

Reply:
Looking at the logic that is in these functions, I would prefer to have something like `GetFabricIDs` that returns a tuple or even a map.

Reply:
is now a single func